File: dixiyang-engine/src/main/java/com/dixiyang/server/Entity/StoryNode.java
package com.dixiyang.server.Entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("story_node")
public class StoryNode {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @TableField("novel_id")
    private Long novelId;

    @TableField("timeline_id")
    private Long timelineId;

    @TableField("title")
    private String title;

    @TableField("content")
    private String content;

    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("vector_id")
    private String vectorId;

    @TableField("event_date")
    private String eventDate;

    @TableField("event_type")
    private String eventType;

    @TableField("importance")
    private Integer importance;

    @TableField("character_names")
    private String characterNames;

    @TableField("tags")
    private String tags;
}

File: DixyangFast/src/dixiyang/schemas/story_node.py
from pydantic import BaseModel, Field


class StoryNodeCreate(BaseModel):
    novel_id: int = Field(alias="novelId")
    timeline_id: int | None = Field(default=None, alias="timelineId")
    title: str
    content: str | None = None
    event_date: str | None = Field(default=None, alias="eventDate")
    event_type: str | None = Field(default=None, alias="eventType")
    importance: int | None = None
    character_names: str | None = Field(default=None, alias="characterNames")
    tags: str | None = None

    model_config = {"populate_by_name": True}


class StoryNodeUpdate(BaseModel):
    timeline_id: int | None = Field(default=None, alias="timelineId")
    title: str | None = None
    content: str | None = None
    event_date: str | None = Field(default=None, alias="eventDate")
    event_type: str | None = Field(default=None, alias="eventType")
    importance: int | None = None
    character_names: str | None = Field(default=None, alias="characterNames")
    tags: str | None = None
